app-frontend/src/app/components/map/annotateSidebarItem/annotateSidebarItem.controller.js

```diff
@@ -0,0 +1,117 @@
+/* globals L, $*/
+
+export default class AnnotateSidebarItemController {
+    constructor(
+        $log, $scope, $location, $anchorScroll, $timeout,
+        mapService
+    ) {
+        'ngInject';
+        this.$log = $log;
+        this.$scope = $scope;
+        this.$location = $location;
+        this.$anchorScroll = $anchorScroll;
+        this.$timeout = $timeout;
+
+        this.getMap = () => mapService.getMap(this.mapId);
+    }
+
+    $onInit() {
+        this.isEditing = false;
+    }
+
+    $onChanges(changeObj) {
+        if (changeObj.scrollId && changeObj.scrollId.currentValue) {
+            this.scrollToItem(changeObj.scrollId.currentValue);
+        }
+    }
+
+    scrollToItem(id) {
+        let newHash = 'anchor' + id;
```

Comment:
This strikes me as logic that could be put elsewhere - this controller should be as small as possible. Logically, this belongs wherever we're managing the list of items, not in each individual item.